twenty-first/src/util_types/merkle_tree.rs

```diff
@@ -394,19 +435,19 @@ impl MerkleTreeInclusionProof {
 impl PartialMerkleTree {
     pub fn root(&self) -> Result<Digest> {
         self.nodes
-            .get(&MerkleTree::ROOT_INDEX)
+            .get(&ROOT_INDEX)
             .copied()
             .ok_or(MerkleTreeError::RootNotFound)
     }
 
-    fn node(&self, index: usize) -> Result<Digest> {
+    fn node(&self, index: MerkleTreeNodeIndex) -> Result<Digest> {
         self.nodes
             .get(&index)
             .copied()
             .ok_or(MerkleTreeError::MissingNodeIndex(index))
     }
 
-    fn num_leafs(&self) -> Result<usize> {
+    fn num_leafs(&self) -> Result<MerkleTreeLeafIndex> {
         if self.tree_height > MAX_TREE_HEIGHT {
             return Err(MerkleTreeError::TreeTooHigh);
         }
```

Comment:
After the `MAX_*` constants are gone, this could become something like:

```rust
1_u64
    .checked_shl(self.tree_height)
    .ok_or(MerkleTreeError::TreeTooHigh)
```